File: libraries/demos/count/count1.cc
#include <iostream>
using namespace std;

#define dout if ( debug ) cout
bool debug;

main( int argc )
{
    debug = ( argc > 1 );

    int paragraph = 1;

    while ( ! cin.eof() )
    {
	int characters = 0;
	int words = 0;
	int lines = 0;

	char buffer [101];

	while
	  ( cin.getline ( buffer, sizeof ( buffer ) ),
	    ! cin.eof() )
	{
	    char * cp = buffer;
	    while ( * cp == ' ' ) ++ cp;

	    if ( * cp == 0 ) break;

	    ++ lines;

	    do
	    {
		++ words;
		while ( * cp != ' ' && * cp ) ++ cp;
		while ( * cp == ' ' ) ++ cp;
	    } while ( * cp );

	    characters += ( cp - buffer );
	    dout << "+ " << buffer << endl;
	    dout << ". " << characters
	         << " " << words
		 << " " << lines << endl;
	}

	if ( lines > 0  )
	{
	    cout << "Paragraph " << paragraph << ": "
		 << lines << " lines, "
		 << words << " words, "
		 << characters << " characters."
		 << endl;

	    ++ paragraph;
	}
    }

    return 1;   // This line can be omitted.
		// It is a test that make count.out
		// works even if count returns an
		// error code.
}

File: libraries/demos/count/count1.py
import sys

debug = ( len ( sys.argv ) > 1 )

def dprint (*args):
    global debug
    if debug:
        for arg in args:
	    print arg,
	print

def read():
    try:
        result = raw_input().rstrip ( '\n' )
    except EOFError:
        result = None
    finally:
        return result

paragraph = 1
characters = 0
words = 0
lines = 0

while True:

    line = read()
    if line == None or line.strip() == '':
	if lines > 0:
	    print "Paragraph", str(paragraph) + ":", \
		 lines, "lines,", \
		 words, "words,", \
		 characters, "characters."

	    characters = 0
	    words = 0
	    lines = 0
	    paragraph += 1

	if line == None:
	    break
	else:
	    continue

    lines += 1
    characters += len ( line )
    dprint ( "+", line )

    line = line.strip()
    line = line.split()
    words += len ( line )

    dprint ( ".", characters, words, lines )

exit ( 1 )  # This line can be omitted.
	    # It is a test that make count.out
	    # works even if count returns an
	    # error code.
